pkg/kapp/cmd/app/delete.go

```diff
@@ -149,11 +149,26 @@ func (o *DeleteOptions) Run() error {
 		return err
 	}
 
+	if !shouldFullyDeleteApp {
+		defer func() {
+			_, numDeleted, _ := app.GCChanges(ctlapp.AppChangesMaxToKeepDefault, nil)
```

Comment:
Scenario when the user partially deletes apps using filters we are garbage collecting once it reaches the threshold of 200 similar to the deploy stage. (let me know if you think it is not an ideal number)